### DISCUSSION TITLE
Is there a way to set the time zone for the hasura server?

### DISCUSSION THREAD

**Discussion by timuckun** [hasura/graphql-engine#7579] [Question] [answered]
Hasura is running on a server where the time zone is in UTC.  The JWT server is generating JWT tokens using the local time zone as issued at and expires at.  As a result when an attempt is made the error "Could not verify JWT: JWTIssuedAtFuture" gets generated.

Is there a way to tell Hasura to set it's internal clock to a particular time zone regardless of the server time?

**Comment by BenoitRanque** [comment, marked as answer]
If your jwt is [standard compliant](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.6), it will be using a numeric value for the iat (issued at) claim.
This numeric value should be a unix epoch and thus not affected by timezones. Check if your JWT issuing server's internal time is correct.

**Reply by timuckun** [reply]
In the development environment I am running both hasura and the database in docker compose but the app is running local on my laptop.  The laptop is set to the local time zone and the servers in the docker compose are in UTC (which would be the case in production).

The JWT is being issued by postgres (using the JWT extension) but I am specifying the start and expiry in the local time zone (which is again that I want).

**Reply by BenoitRanque** [reply]
Are you able to confirm that the expiry time being set is what you expect?
Meaning, if you decode the JWT payload and take a look at the `iat` claim, is it the value you expect?

**Reply by timuckun** [reply]
Thanks for the suggestion.  

It turns out that the postgres JWT function takes the time you give it and treats it as if was the UTC time. So if I say 5:00 PM at time zone 'My/time'  the JWT gets created at 5:00 PM UTC time.  So the timestamp indeed gets set in the future.

**Reply by BenoitRanque** [reply]
You should be able to either send a timestamp with a time zone, or an integer to offset the lifetime by. Glad you found the root of the problem, I will mark this as solved.

**Reply by timuckun** [reply]
That's the weird thing. I was sending a timestamp with a time zone but it was somehow ignoring the timezone and pretending it was UTC.
